### PR TITLE
bugfix: fix(#448) redis pool with config error

### LINKED ISSUE #448: redis pool functional options has bug
**Describe the bug**
```golang
// WithConfig set new config for NewPool,keep old code compatibility
func WithConfig(newConfig *Config) Option {
	return func(c *Config) {
		c = newConfig
	}
}
```

**To Reproduce**
```golang
func TestWithConfig(t *testing.T) {
	baseConfig := &Config{}
	newConfig := &Config{DB: 16}
	options := []Option{
		WithConfig(newConfig),
	}
	for _, option := range options {
		option(baseConfig)
	}
	assert.Equalf(t, baseConfig.DB, newConfig.DB, "with config failed")
}

```

The testcase we got:

```
Expected :0
Actual   :16
```

------

To solve this:
1. Or remove WithConfig option 
2. Or copy each field 
3. Or `redisClient := NewRedisClient(config, configOpts...)`
4. Or use deep copy

### PR DESCRIPTION
**Pull request by edocevol**
**Please provide issue(s) of this PR:**
Fixes #448 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [x] Configuration
- [ ] Docs
- [ ] Installation
- [x] Performance and Scalability
- [ ] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: common/redispool/config_option.go [hunk 1]
```diff
@@ -29,7 +31,7 @@ type Option func(c *Config)
 // WithConfig set new config for NewPool,keep old code compatibility
 func WithConfig(newConfig *Config) Option {
 	return func(c *Config) {
-		c = newConfig
+		_ = copier.CopyWithOption(c, newConfig, copier.Option{IgnoreEmpty: true, DeepCopy: true})
```
Comment (chuntaojun) [inline common/redispool/config_option.go:34]: just *c = *newConfig is ok

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: common/redispool/config_option.go (hunk 1 above)
Comment (edocevol) [reply common/redispool/config_option.go]: Great!

**Review by chuntaojun — APPROVED** [review]

**Review by chuntaojun — APPROVED** [review]

**Comment by daheige** [comment]
Here is to overwrite the value, it will be processed automatically, no * is needed, please see the unit test case and the newpool usage of Polaris.
This is not a bug.

**Comment by edocevol** [comment]
https://github.com/polarismesh/polaris/pull/454

**Pull request closed without merging**